assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -63,6 +76,19 @@ function MediaProvider({ children }) {
 
   useEffect(fetchMedia, [fetchMedia, mediaType, searchTerm]);
 
+  const generatePoster = useCallback(() => {
+    const processor = async ({ mimeType, posterId, id, src }) => {
+      if (allowedVideoMimeTypes.includes(mimeType) && !posterId) {
+        await uploadVideoFrame(id, src);
+      }
+    };
+    if (media) {
+      media.forEach(processor);
```

Comment:
Should we debounce here? At the moment, the poster generate does wait to complete before I new one is started, but might better to just add a little wait in there. Make sure the editor is responsive. Thoughts @miina @merapi ?

Reply:
Makes sense to me!

Reply:
I think doing this for every element is definitely a lot. Why not just wait for the visibility signal from the infinite scroller, and/or hover/focus signals? Even with those we'll still want to debounce most likely.

Reply:
I dont think this is a issue. All we want to do here, is just process all the videos in the library. This is only ever run one time. Once done, we need not worry about this.

Reply:
This could still be a big load on the system. At the very least we should think about debouncing them then. What if we simplify this now via `requestIdleCallback`?